control-base/src/main/java/fi/nls/oskari/control/feature/InsertFeatureHandler.java

```diff
@@ -125,5 +126,21 @@ private String parseFeatureIdFromResponse(String response)
             throw new ActionException("SAX processing error", ex);
         }
     }
+
+    /**
+     * Takes workspace prefix from layer name (before ':') and puts it into the layer url after 'geoserver/'
+     * @param  layer  OskariLayer layer
+     */
+    private void movePrefixFromNameToURL(OskariLayer layer){
+        String layerName = layer.getName();
+        String url = layer.getUrl();
+        if(layerName.indexOf(":") != -1){
+            url = url.replace("geoserver/","geoserver/"+(layerName.split(":")[0])+"/");
```

Comment:
This is a very specific thing to do. First of all it expects geoserver/ to be in the url. Then with an example layer name of oskari:userlayer
- URL: https://something/geoserver/ows would be changed to https://something/geoserver/oskari/ows
- URL: https://something/geoserver/oskari/wfs would be changed to https://something/geoserver/oskari/oskari/wfs

Maybe this needs some refining?